File: back/post-service/src/main/java/com/elducche/postservice/service/ThemeService.java
package com.elducche.postservice.service;

import com.elducche.postservice.models.Theme;
import com.elducche.postservice.repositories.ThemeRepository;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.Optional;

@Service
public class ThemeService {

    private final ThemeRepository themeRepository;

    public ThemeService(ThemeRepository themeRepository) {
        this.themeRepository = themeRepository;
    }

    /**
     * Récupère tous les thèmes
     */
    public List<Theme> getAllThemes() {
        return themeRepository.findAll();
    }

    /**
     * Récupère un thème par son ID
     */
    public Optional<Theme> getThemeById(Long id) {
        return themeRepository.findById(id);
    }

    /**
     * Crée un nouveau thème
     */
    public Theme createTheme(Theme theme) {
        return themeRepository.save(theme);
    }
}

File: front/angular-app/src/app/shared/services/theme.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ConfigService } from '../../core/services/config.service';
import { Theme } from '../interfaces/theme.interface';

@Injectable({
  providedIn: 'root'
})
export class ThemeService {
  private readonly http = inject(HttpClient);
  private readonly config = inject(ConfigService);

  /**
   * Récupère tous les thèmes disponibles
   */
  getAllThemes(): Observable<Theme[]> {
    return this.http.get<Theme[]>(this.config.endpoints.themes.all);
  }

  /**
   * Récupère un thème par son ID
   */
  getThemeById(id: number): Observable<Theme> {
    return this.http.get<Theme>(`${this.config.apiUrl}/api/themes/${id}`);
  }
}
